### ISSUE TITLE
BUG : GroupTraits do not support PostHog's required groupType field

### ISSUE THREAD

**Issue by JJCLane** [rudderlabs/rudder-sdk-flutter#104]
**Describe the bug**
Please provide the following information:

1. Following the integration outlined here: https://www.rudderstack.com/docs/destinations/streaming-destinations/posthog/posthog-cloud-mode/#updated-group-mapping
2. It's currently not possible to specify a group type for PostHog group events when using the `rudder-sdk-flutter` package due to the limitations of the `RudderTraits` model.

**Which platform is the issue occurring on**
- [x] Web, Android and iOS

**Expected behavior**
Either RudderStack will promote traits in the `extra` property, or the `RudderTraits` model has the ability to set arbitrary values which don't live in the `extra` field. I tested locally and setting it outside of the `extra` field works as expected.

**To Reproduce**
Steps to reproduce the behaviour:

1. Initialise Flutter SDK
2. Send a group event like so:
```dart
RudderTraits groupTraits = RudderTraits();
groupTraits.put("groupType", "organization");
groupTraits.put("name", "My Org");
RudderController.instance.group('my-org, groupTraits: groupTraits);
```
3. See this event gets fired, but with the `groupType` key contained within the `extras` key:
![image](https://github.com/rudderlabs/rudder-sdk-flutter/assets/3543345/459a90e1-7631-41a4-8a6f-dd2bbaa007c0)
4. Observe within RudderStack cloud, this event gets forwarded to PostHog, but without the necessary `"$group_type"` field and therefor doesn't work:
```json
{
    "body": {
        "XML": {},
        "FORM": {},
        "JSON": {
            "type": "group",
            "event": "$groupidentify",
            "api_key": "",
            "messageId": "831c3d0b-336a-4e22-8740-2170babf46a3",
            "timestamp": "2023-07-21T23:06:28.508Z",
            "properties": {
                "$ip": "",
                "$os": "Mac OS",
                "$lib": "RudderLabs JavaScript SDK",
                "$set": {},
                "$host": "localhost:65303",
                "$time": "2023-07-21T23:06:28.508Z",
                "$locale": "en-US",
                "$browser": "Chrome",
                "$app_name": "RudderLabs JavaScript SDK",
                "$pathname": "/",
                "$group_key": "my-org",
                "$group_set": {
                    "extras": {
                        "name": "My Org",
                        "groupType": "organization",
                    }
                },
                "$timestamp": "2023-07-21T23:06:28.508Z",
                "$useragent": "",
                "distinct_id": "9f58fbfb-c0b5-4bd2-8580-1c5d46fc37d8",
                "$app_version": "1.33.0",
                "$current_url": "http://localhost:65303/",
                "$lib_version": "1.33.0",
                "$screen_width": 2560,
                "$app_namespace": "com.rudderlabs.javascript",
                "$screen_height": 1440,
                "$screen_density": 1,
                "$browser_version": "114.0.0.0",
                "$anon_distinct_id": "9f58fbfb-c0b5-4bd2-8580-1c5d46fc37d8"
            },
            "distinct_id": "9f58fbfb-c0b5-4bd2-8580-1c5d46fc37d8"
        },
        "JSON_ARRAY": {}
    },
    "type": "REST",
    "files": {},
    "method": "POST",
    "params": {},
    "userId": "",
    "headers": {
        "Content-Type": "application/json"
    },
    "version": "1",
    "endpoint": "https://app.posthog.com/batch"
}
```

**Screenshots**

**Version of the _Flutter_ SDK**
Please provide the following information:

1. The version of the Rudder Flutter SDK you are using (Rudder Flutter SDK v2.3.0)
3. The versions of the Rudder Android and iOS SDKs utilized within Flutter (e.g., Rudder Android SDK v2.3.0 and Rudder iOS SDK v2.3.0)

**SDK initialisation snippet**
```dart
final RudderController controller = RudderController.instance;
    RudderLogger.init(RudderLogger.VERBOSE);
    RudderConfigBuilder builder = RudderConfigBuilder();
    builder.withDataPlaneUrl("https://{tenant}.dataplane.rudderstack.com");
    builder.withMobileConfig(MobileConfig(recordScreenViews: true));
    controller.initialize("{API_KEY}",
        config: builder.build());
```

**Comment by desusai7** [comment]
Hey @JJCLane, 

Thank you for bringing this to our notice, we just raised a PR to fix this, we will have it released in some time, will inform you once released

**Comment by JJCLane** [comment]
Amazing, thanks for the crazy fast turnaround @desusai7!

**Comment by desusai7** [comment]
Hey @JJCLane, 

We released version 2.4.1 of the SDK yesterday with the fix for this issue, can you please upgrade to this version and let us know how this goes ?
